server/service/queries.go

```diff
@@ -516,6 +534,23 @@ func (svc *Service) ApplyQuerySpecs(ctx context.Context, specs []*fleet.QuerySpe
 				Message: fmt.Sprintf("query payload verification: %s", err),
 			})
 		}
+
+		dbQuery, err := svc.ds.QueryByName(ctx, query.TeamID, query.Name)
+		if err != nil && !fleet.IsNotFound(ctxerr.Unwrap(err)) {
+			return ctxerr.Wrap(ctx, fleet.NewBadGatewayError(fmt.Sprintf("fetching saved query"), err))
+		}
+
+		if dbQuery == nil {
+			// then we're creating a new query, so move on.
+			continue
+		}
+
+		if query.DiscardData || query.Logging != fleet.LoggingSnapshot || query.Query != dbQuery.Query {
+			err = svc.ds.DeleteAllResultsForQueryByName(ctx, query.Name)
```

Comment:
Let's delete by ID, you already have `dbQuery.ID`. (Because you can have two queries with the same name that belong to different teams.)

Reply:
Ah gotcha. I thought names were globally unique, my b!